### ISSUE TITLE
MIR borrowck: spurious error with ref pattern in argument

### ISSUE THREAD

**Issue by arielb1** [rust-lang/rust#46161]
e.g. (or src/test/run-pass/issue-8860.rs)
```Rust
fn main() {}
fn foo(ref x: Box<()>) {}
```

The problem is that region inference infers the borrow of `x` to `ReEmpty` (because it's live nowhere), which doesn't have an EndRegion. NLL mode (`-Z nll`) is not affected by this particular pathology, but I think this should still be fixed.
